c10/core/TensorImpl.h

```diff
@@ -138,6 +138,68 @@ struct C10_API AutogradMetaInterface {
   virtual ~AutogradMetaInterface();
 };
 
+// NOTE [ Version Counter Sharing ]
+//
+// Every Tensor has a version counter. Version counters are incremented
+// whenever the data or shape of a tensor changes through Variable operations.
+// These are typically in-place operations. Version counters are used to
+// detect modifications to saved variables which would result in incorrect
+// gradient calculations. Version counters may be shared between Variables:
+//
+// 1. A view shares the version counter of the base Variable,
+// 2. Detached variables share the version counter of the source,
```

Comment:
It's important to distinguish `x.detach()` from `x.data`. The former shares version counter; the latter doesn't, am I right? You should actually write code here. A link to #5396 would make it even better!